### PR TITLE
Raise a ResponseError for FedEx#find_tracking_info

### PR DESCRIPTION
**Pull request by thegedge**
In a couple of places we were raising an `ActiveShipping::Error` when we could raise/return a more informed `ActiveShipping::TrackingResponse(false, ...)` that has additional information (e..g, response, request) for clients of this gem.

The one point of discussion here is w.r.t. the semantics of `ResponseError`. Generally it means there was an issue in the response, but for these two cases I've changed aren't considered error responses by FedEx, and aren't given special status codes (i.e., `/TrackReply/Notifications/Code` is 0), so perhaps this doesn't make sense?

Also made a few small formatting changes on nearby code (separate commit).

### CODE REVIEW

File: test/unit/carriers/fedex_test.rb [hunk 1]
```diff
@@ -327,6 +327,23 @@ def test_failure_to_parse_invalid_xml_results_in_a_useful_error
...
+  def test_parsing_response_without_notifications
+    mock_response = xml_fixture('fedex/reply_without_notifications')
+
+    @carrier.expects(:commit).returns(mock_response)
+
+    response = @carrier.find_rates(
+      location_fixtures[:ottawa],
+      location_fixtures[:beverly_hills],
+      package_fixtures.values_at(:book, :wii),
+      :test => true
```
Comment (jonathankwok) [inline test/unit/carriers/fedex_test.rb]: `test: true` would be 👍  here.

File: test/unit/carriers/fedex_test.rb [hunk 2]
```diff
@@ -327,6 +327,23 @@ def test_failure_to_parse_invalid_xml_results_in_a_useful_error
...
+    )
+
+    assert response.success?
```
Comment (jonathankwok) [inline test/unit/carriers/fedex_test.rb]: `assert_predicate`

File: test/unit/carriers/fedex_test.rb [hunk 3]
```diff
@@ -507,14 +507,25 @@ def test_tracking_info_without_status
...
-    error = assert_raises(ActiveShipping::Error) do
+    error = assert_raises(ActiveShipping::ResponseError) do
       @carrier.find_tracking_info('123456789012')
     end
 
     msg = 'Multiple matches were found. Specify a unqiue identifier: 2456987000~123456789012~FX, 2456979001~123456789012~FX, 2456979000~123456789012~FX'
     assert_equal msg, error.message
   end
 
+  def test_tracking_info_without_tracking_details
```
Comment (jonathankwok) [inline test/unit/carriers/fedex_test.rb]:
```
test "#find_tracking_info without tracking details" do
```

Comment (jonathankwok) [inline test/unit/carriers/fedex_test.rb]: You could also change it throughout this PR. Or, keep it as is and have a dedicated PR to updating this test file to `test "" do`

Comment (thegedge) [inline test/unit/carriers/fedex_test.rb]: Yeah, I'd rather keep that as a bigger cleanup PR to keep things cohesive.

**Comment by thegedge** [comment]
Updated

**Comment by thegedge** [comment]
Okay, I added one more commit which was the actual real issue that needed to be dealt with here, but we can keep the `TrackingResponse` code here too.

tl;dr we were only looking at the success status from the document root, and in a `TrackDetails` element, but forgetting to check `CompleteTrackDetails`. I've updated the test to capture this case.

**Comment by thegedge** [comment]
^ Squashed it into two, cohesive commits that I'll merge (they don't make sense to merge together)
